src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -673,7 +673,10 @@ private void maybeGossipToUnreachableMember(MessageOut<GossipDigestSyn> message)
             double prob = unreachableEndpointCount / (liveEndpointCount + 1);
             double randDbl = random.nextDouble();
             if (randDbl < prob)
-                sendGossip(message, unreachableEndpoints.keySet());
+                // Theoritically justRemovedEndpoints and unreachableEndpoints should have no intersections
+                // but the Gossiper is prone to races
+                sendGossip(message, Sets.filter(unreachableEndpoints.keySet(),
```

Comment:
can we do this instead
```
sendGossip(message, Sets.difference(unreachableEndpoints.keySet(), justRemovedEndpoints.keySet()));
```

Reply:
Discussed offline, keeping filter as it's closer to pattern in later version of C*: https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/gms/Gossiper.java#L807